src/feat/pitch-functions.cc

```diff
@@ -835,19 +844,35 @@ void OnlinePitchFeatureImpl::ExtractFrame(
   int32 offset = static_cast<int32>(sample_index -
                                     downsampled_samples_processed_);
 
+  // Treat edge cases first
+  if (sample_index < 0) {
+    // Part of the frame is before the beginning of the signal. This
+    // should only happen if opts_.snip_edges == false, when we are
+    // processing the first few frames of signal. In this case
+    // we pad with zeros.
+    KALDI_ASSERT(opts_.snip_edges == false);
```

Comment:
Can you please re-test?  This is a change that would break a lot of setups
in a way that would be hard to roll back, if it was wrong.


On Tue, Jun 21, 2016 at 7:13 PM, Blaise Potard <notifications@github.com>
wrote:

> In src/feat/pitch-functions.cc
> <https://github.com/kaldi-asr/kaldi/pull/823#discussion_r67969538>:
>
> > @@ -835,19 +844,35 @@ void OnlinePitchFeatureImpl::ExtractFrame(
> >    int32 offset = static_cast<int32>(sample_index -
> >                                      downsampled_samples_processed_);
> >
> > +  // Treat edge cases first
> > +  if (sample_index < 0) {
> > +    // Part of the frame is before the beginning of the signal. This
> > +    // should only happen if opts_.snip_edges == false, when we are
> > +    // processing the first few frames of signal. In this case
> > +    // we pad with zeros.
> > +    KALDI_ASSERT(opts_.snip_edges == false);
>
> I had tested before my initial commit on some tests file that the output
> were identical to the current git with snip-edges=true; I have no retested
> after the recent changes but I would not expect any change in behaviour.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/823/files/6a9b8a25c9129b57b84560cd60d2c9b0bd9e806b#r67969538>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADJVu1lV5TRgf_Yz-vBsgB3_ba0zK_-zks5qOHAWgaJpZM4IqgEB>
> .
>